backend/api/Services/DocumentService.cs

```diff
@@ -95,8 +96,17 @@ public async Task<DocumentUploadResponse> UploadActivityDocumentAsync(long activ
             DocumentUploadResponse response = new DocumentUploadResponse() { ExternalResult = externalResult };
             if (externalResult.Status == ExternalResultStatus.Success)
             {
-                // Create the pims document
                 var externalDocument = externalResult.Payload;
+
+                // Save metadata of document
+                IList<Task<ExternalResult<DocumentMetadata>>> metadataTasks = new List<Task<ExternalResult<DocumentMetadata>>>();
+                foreach (var metadata in uploadRequest.DocumentMetadata)
+                {
+                    metadataTasks.Add(documentStorageRepository.CreateDocumentMetadataAsync(externalDocument.Id, metadata.MetadataTypeId, metadata.Value));
+                }
+                Task.WaitAll(metadataTasks.ToArray());
```

Comment:
I think this is preferred is it not? await Task.WhenAll(metadataTasks.ToArray()); to avoid blocking the thread.

Reply:
done